projects/hopr/index.js

```diff
@@ -0,0 +1,71 @@
+const sdk = require("@defillama/sdk");
+const { getLogs } = require('../helper/cache/getLogs')
+
+const wxHOPR_TOKEN_SMART_CONTRACT_ADDRESS = '0xD4fdec44DB9D44B8f2b6d529620f9C0C7066A2c1';
+const xHOPR_TOKEN_SMART_CONTRACT_ADDRESS  = '0xD057604A14982FE8D88c5fC25Aac3267eA142a08';
+const SAFE_FACTORY = '0x098B275485c406573D042848D66eb9d63fca311C';
+const HOPR_CHANNELS = '0x693Bac5ce61c720dDC68533991Ceb41199D8F8ae';
+const TOKEN_DECIMALS = 18;
+const CHAIN = 'xdai';
+
+function tvl() {
+  return async (timestamp, ethBlock, chainBlocks, { api }) => {
+
+    const logs = await getLogs({
+      api,
+      target: SAFE_FACTORY,
+      topic: 'NewHoprNodeStakeSafe(address)',
+      fromBlock: 29706820,
+      eventAbi: 'event NewHoprNodeStakeSafe(address instance)'
+    });
+
+    const safes = logs.map(log => log.args[0]);
+    const balanceOfSafes = await balanceOfs(wxHOPR_TOKEN_SMART_CONTRACT_ADDRESS, CHAIN, safes);
+    const balanceOfChannels = BigInt((
+      await sdk.api.abi.call({
+        abi: 'erc20:balanceOf',
+        target: wxHOPR_TOKEN_SMART_CONTRACT_ADDRESS,
+        params: HOPR_CHANNELS,
+        chain: CHAIN
+      })
+    ).output);
+
+    const BigIntTvl = balanceOfSafes + balanceOfChannels;
+
+    const tvl = { [`xdai:${xHOPR_TOKEN_SMART_CONTRACT_ADDRESS}`]: BigIntTvl.toString() }
```

Comment:
Why does the key use `xHOPR_TOKEN_SMART_CONTRACT_ADDRESS`?

Reply:
Key uses the xHOPR because here we choose the token for the defi.lama to count the value in USD of TVL.  wxHOPR has no value